pkg/cloud/services/ec2/ami.go

```diff
@@ -114,7 +118,13 @@ func (s *Service) pickArchitectureForInstanceType(instanceType string) (string,
 	}
 	describeInstanceTypeResult, err := s.EC2Client.DescribeInstanceTypes(descInstanceTypeInput)
 	if err != nil {
-		return "", err
+		// if call to DescribeInstanceTypes fails due to permissions error, log a warning and return the default architecture.
+		if awserrors.IsPermissionsError(err) {
+			record.Warnf(s.scope.InfraCluster(), "FailedDescribeInstanceTypes", "insufficient permissions to describe instance types for instance type %q: %v", instanceType, err)
```

Comment:
It would be nice in the warning to mention that we are falling back to the default architecture of x86_64.